sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategySuite.scala

```diff
@@ -27,18 +27,18 @@ import org.apache.spark.sql.types.{IntegerType, StringType, StructField, StructT
 class DataSourceStrategySuite extends PlanTest with SharedSparkSession {
   val attrInts = Seq(
     $"cint".int,
-    $"c.int".int,
+    $"`c.int`".int,
```

Comment:
The AttributeReferences created from these were previously treated as single-part which is why I'm updating the test to actually make these single-part (according to the comments on the test that's the expected behaviour here).

We could decide to treat `$"...".int` as single-part instead, which maybe is less of a change compared to the current behavior. But other tests rely on producing multi-part identifiers with `$"left.a"` [(here)](https://github.com/apache/spark/blob/87a5442f7ed96b11051d8a9333476d080054e5a0/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/ReplaceOperatorSuite.scala#L180).